support/modules/rootfs-image

```diff
@@ -44,20 +25,55 @@ else
     SETENV=fw_setenv
 fi
 
-active_num="$(${PRINTENV} mender_boot_part)"
-active_num="${active_num#mender_boot_part=}"
-if test $active_num -eq $MENDER_ROOTFS_PART_A_NUMBER; then
-    active=$MENDER_ROOTFS_PART_A
-    passive=$MENDER_ROOTFS_PART_B
-    passive_num=$MENDER_ROOTFS_PART_B_NUMBER
-else
-    active=$MENDER_ROOTFS_PART_B
-    passive=$MENDER_ROOTFS_PART_A
-    passive_num=$MENDER_ROOTFS_PART_A_NUMBER
-fi
-passive_num_hex=$(printf '%x' $passive_num)
-upgrade_available="$(${PRINTENV} upgrade_available)"
-upgrade_available="${upgrade_available#upgrade_available=}"
+parse_conf_file() {
+    found_conf_file=
+    # Try first the fallback config file, which has least precedence
+    if grep -q RootfsPartA "${MENDER_DATASTORE_DIR:-/var/lib/mender}/mender.conf" 2> /dev/null; then
+        CONF_FILE="${MENDER_DATASTORE_DIR:-/var/lib/mender}/mender.conf"
+        found_conf_file=yes
+    fi
+    if grep -q RootfsPartA "${MENDER_CONF_DIR:-/etc/mender}/mender.conf" 2> /dev/null; then
```

Comment:
This is ok, but wouldn't it be both easier and more robust to slap a loop around the parsing below, once over `/data`, once over `/etc`? You wouldn't even need to check a condition then, you can just use something like:

```
tmp="$(jq -r .RootfsPartA < "$CONF_FILE")"
MENDER_ROOTFS_PART_A="${tmp:-${MENDER_ROOTFS_PART_A}}"
```

An advantage then is that if `jq` is installed, it's guaranteed to be parsed correctly, whereas this block can still misparse `RootfsPartA` if it happens to be in a comment for example. Not that I expect it, but you know. 🙂

Just make sure that the `jq` and `sed` calls use ` || true` to avoid `set -e` kicking in.

Reply:
`json` does not allow comments. But in any case what you say makes sense. You could have a `RootfsPartAA` key or `RootfsPartA` as part of a tenant token (I guess it is technically _possible_ :laughing: ).

I'll clean it up, thanks.